test/sample_exact_es_percent/processor-spec.ts

```diff
@@ -0,0 +1,236 @@
+
+        expect(results.length).toEqual(9500);
+    });
+
+    it('with large datasets and 50%', async () => {
+        percentArr = [50];
+        const data = makeData(10000);
+        harness = await makeHarness();
+        const results = await harness.runSlice(data);
+
+        expect(results.length).toEqual(5000);
+    });
+
+    it('updates percentage every window_ms', async () => {
+        percentArr = [50, 0, 100];
+        const data = makeData(10000);
+        harness = await makeHarness({ window_ms: 100, index: 'my-index', document_id: 'abc123' });
+        const results1 = await harness.runSlice(data);
+        await pDelay(100);
+        const results2 = await harness.runSlice(data);
+        await pDelay(100);
+        const results3 = await harness.runSlice(data);
+
+        expect(results1.length).toEqual(5000);
+        expect(results2.length).toEqual(0);
+        expect(results3.length).toEqual(10000);
+    }, 30000);
+
+    it('percentage can be number or string', async () => {
+        percentArr = ['50', 0, '100'];
+        const data = makeData(10000);
+        harness = await makeHarness({ window_ms: 100, index: 'my-index', document_id: 'abc123' });
+        const results1 = await harness.runSlice(data);
+        await pDelay(100);
+        const results2 = await harness.runSlice(data);
+        await pDelay(100);
+        const results3 = await harness.runSlice(data);
+
+        expect(results1.length).toEqual(5000);
+        expect(results2.length).toEqual(0);
+        expect(results3.length).toEqual(10000);
+    }, 30000);
+
+    describe('-> _getNewPercentKept errors', () => {
+        // client will shift off and return first element of docArr[].
+        // Each request will be tried 3 times, so 3 copies of any document that throws are required.
+        let docArr: { found: boolean; _source: object }[] = [];
+        const errorClients: TestClientConfig[] = [
+            {
+                type: 'elasticsearch-next',
+                endpoint: 'default',
+                createClient: async () => ({
+                    client: {
+                        get: () => {
+                            return docArr.shift();
+                        }
+                    },
+                    logger
+                }),
+            }
+        ];
+
+        it('should throw if initial request fails', async () => {
+            const doc = { found: false };
+            docArr = Array(3).fill(doc);
+            await expect(makeHarness({ index: 'my-index', document_id: 'abc123' }, errorClients)).rejects
+                .toThrow('SampleExactESPercentage failed to retrieve percentage from index my-index of '
+                    + 'elasticsearch-next connection default: TSError: The document with id abc123 was '
+                    + 'not found in index my-index of elasticsearch-next connection default.');
+        });
+
+        it('should throw if document not found', async () => {
```

Comment:
Isn't this identical to the test above: `'should throw if initial request fails'`

Am I missing a difference?